tests/sentry/api/endpoints/test_organization_details.py

```diff
@@ -807,6 +811,84 @@ def test_org_mapping_already_taken(self):
 
+    def test_configure_auth_provider(self):
+        with self.feature("organizations:api-auth-provider"):
+            old_config = {"domain": "foo.com"}
+            new_config = {"domain": "bar.com"}
+            provider = "google"
+
+            self.get_success_response(
+                self.organization.slug,
+                method="put",
+                providerKey=provider,
+                providerConfig=old_config,
+            )
+            auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+            assert auth_provider.provider == provider
+            assert auth_provider.config == {
+                "domains": ["foo.com"],
+                "version": DATA_VERSION,
+                "sentry-source": "api-organization-details",
+            }
+
+            self.get_success_response(
+                self.organization.slug,
+                method="put",
+                providerKey=provider,
+                providerConfig=new_config,
+            )
+            auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+            assert auth_provider.provider == provider
+            assert auth_provider.config == {
+                "domains": ["bar.com"],
+                "version": DATA_VERSION,
+                "sentry-source": "api-organization-details",
+            }
+
+    def test_invalid_auth_provider_configuration(self):
+        with self.feature("organizations:api-auth-provider"):
+            self.get_error_response(
+                self.organization.slug, method="put", providerKey="google", status_code=400
+            )
+            self.get_error_response(
+                self.organization.slug,
+                method="put",
+                providerConfig={"domain": "foo.com"},
+                status_code=400,
+            )
+            self.get_error_response(
+                self.organization.slug,
+                method="put",
+                providerKey="not_valid",
+                providerConfig={"domain": "foo.com"},
+                status_code=400,
+            )
+
+            response = self.get_error_response(
+                self.organization.slug,
+                method="put",
+                providerKey="google",
+                providerConfig={"invalid_domain": "foo.com"},
+                status_code=400,
+            )
+            assert response.data == {
+                "providerConfig": ["Invalid key 'domain' in providerConfig for authprovider google"]
+            }
+
+        # Error if passing in providerKey or providerConfig and feature flag has not been set
+        self.get_error_response(
+            self.organization.slug,
+            method="put",
+            providerKey="google",
+            status_code=400,
+        )
+        self.get_error_response(
+            self.organization.slug,
+            method="put",
+            providerConfig={"option": "test"},
+            status_code=400,
+        )
```

Comment:
This is not a good test. 

If you're testing the feature flag, you can just combine into 1 valid request with both `providerKey` and `providerConfig` that becomes invalid due to the missing flag.

Otherwise, the error might be due missing keys and not necessarily the missing feature flag.

Reply:
Also, I rather you break this out into it's own separate test.